crates/oxc_query/src/properties.rs

```diff
@@ -0,0 +1,584 @@
+                })
+                .import
+                .source
+                .value
+                .to_string()
+                .into()
+        }),
+        _ => {
+            unreachable!("attempted to read unexpected property '{property_name}' on type 'Import'")
+        }
+    }
+}
+
+pub(super) fn resolve_import_ast_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "from_path" => resolve_import_property(contexts, property_name, resolve_info),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'ImportAST'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_interface_extend_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "str" => interface_extend_implem(contexts),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'InterfaceExtend'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_jsxattribute_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "name" => resolve_property_with(contexts, |v| {
+            jsx_attribute_name_to_string(
+                &v.as_jsx_attribute()
+                    .unwrap_or_else(|| {
+                        panic!("expected to have a jsxattribute vertex, instead have: {v:#?}")
+                    })
+                    .name,
+            )
+            .into()
+        }),
+        "value_as_constant_string" => resolve_property_with(contexts, |v| {
+            let attr = v.as_jsx_attribute().unwrap_or_else(|| {
+                panic!("expected to have a jsxattribute vertex, instead have: {v:#?}")
+            });
+            jsx_attribute_to_constant_string(attr).map_or_else(|| FieldValue::Null, Into::into)
+        }),
+        _ => {
+            unreachable!(
+                "attempted to read unexpected property '{property_name}' on type 'JSXAttribute'"
+            )
+        }
+    }
+}
+
+pub(super) fn resolve_jsxelement_property<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    property_name: &str,
+    _resolve_info: &ResolveInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, FieldValue> {
+    match property_name {
+        "as_constant_string" => resolve_property_with(contexts, |v| {
+            v.as_constant_string().map_or(FieldValue::Null, Into::into)
+        }),
+        "child_count" => resolve_property_with(contexts, |v| {
```

Comment:
is this bad api? we give a bunch of iterators, one for each type of child, but none for all the children, so instead we just provide a total child count

Reply:
Depends, impossible to say in a vacuum. When would someone want an iterator over all children? When would someone want the total child count?

Reply:
I don't think they would, given the opportunities for more specific accessors, thus I think this is the right API